extensions/android-adb-input/src/index.tsx

```diff
@@ -0,0 +1,157 @@
+import {
+  Action,
+  ActionPanel,
+  Detail,
+  Form,
+  getPreferenceValues,
+  popToRoot,
+  PreferenceValues,
+  showHUD,
+  showToast,
+  Toast,
+} from "@raycast/api";
+import { useState, useEffect } from "react";
+import { execaCommand } from "execa";
+import * as fs from "fs";
+
+type Error = {
+  title: string;
+  message: string;
+};
+
+function getAdbDir(): string {
+  const preferences = getPreferenceValues<PreferenceValues>();
+  const adbDir = preferences["adbDir"];
+  if (adbDir) {
+    return adbDir;
+  }
+
+  const home = process.env.HOME;
+  return `${home}/Library/Android/sdk/platform-tools`;
+}
+
+function existsAdb(): boolean {
+  return fs.existsSync(`${getAdbDir()}/adb`);
+}
+
+async function getDeives(): Promise<string[]> {
+  const adbDir = getAdbDir();
+  const { stdout } = await execaCommand(`${adbDir}/adb devices`, { cwd: adbDir });
+
+  const devices: string[] = [];
+
+  stdout.split("\n").forEach((line, index) => {
+    if (index != 0 && line.length > 0) {
+      devices.push(line.split("\t")[0]);
+    }
+  });
+  return devices;
+}
+
+function onSubmit(values: Form.Values) {
+  const device = values["device"];
+  const text = values["text"];
+
+  if (!text) {
+    showToast({
+      title: "Error",
+      message: "Text is empty.",
+      style: Toast.Style.Failure,
+    });
+    return;
+  }
+  if (!device) {
+    showToast({
+      title: "Error",
+      message: "Device is not selected.",
+      style: Toast.Style.Failure,
+    });
+    return;
+  }
+
+  const excapedText = text.replace(" ", "%s").replace("\\", "\\\\");
+  const args = `-s ${device} shell input text "${excapedText}"`;
+
+  const adbDir = getAdbDir();
+
+  showToast({
+    title: "Sending...",
+    style: Toast.Style.Animated,
+  });
+
+  execaCommand(`${adbDir}/adb ${args}`, { cwd: adbDir })
+    .then(() => {
+      popToRoot({ clearSearchBar: true });
+      showHUD("Complete!");
+    })
+    .catch((e) => {
+      showToast({
+        title: "Error",
+        message: e.message,
+        style: Toast.Style.Failure,
+      });
+    });
+}
+
+export default function Command() {
+  const [devices, setDevices] = useState<string[]>([]);
+  const [error, setError] = useState<Error>();
+
+  useEffect(() => {
+    if (!existsAdb()) {
+      setError({
+        title: "adb command not found!",
+        message: "Please check the adb directory path. See `Raycast Preferences > Extensions > Run Android ADB Input`.",
+      });
+      return;
+    }
+
+    (async () => {
+      const toast = await showToast({
+        title: "Search devices...",
+        style: Toast.Style.Animated,
+      });
+
+      try {
+        const devices = await getDeives();
```

Comment:
```suggestion
        const devices = await getDevices();
```
Totally NIT but could be nice to have corrected 😄

Reply:
Fix [a5f1062](https://github.com/raycast/extensions/pull/964/commits/a5f106229b7eee34278b237abe5ce7f24fd4aa8b)